# Titanic competition with TensorFlow Decision Forests

This notebook will take you through the steps needed to train a baseline Gradient Boosted Trees Model using TensorFlow Decision Forests and creating a submission on the Titanic competition. 

This notebook shows:

1. How to do some basic pre-processing. For example, the passenger names will be tokenized, and ticket names will be splitted in parts.
1. How to train a Gradient Boosted Trees (GBT) with default parameters
1. How to train a GBT with improved default parameters
1. How to tune the parameters of a GBTs
1. How to train and ensemble many GBTs

# Imports dependencies

In [1]:
import numpy as np
import pandas as pd
import os

import tensorflow as tf
import tensorflow_decision_forests as tfdf

print(f"Found TF-DF {tfdf.__version__}")

Found TF-DF 1.2.0


# Load dataset

In [2]:
train_df = pd.read_csv("/kaggle/input/titanic/train.csv")
serving_df = pd.read_csv("/kaggle/input/titanic/test.csv")

train_df.head(10)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   
5            6         0       3   
6            7         0       1   
7            8         0       3   
8            9         1       3   
9           10         1       2   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   
5                                   Moran, Mr. James    male   NaN      0   
6                            McCarthy, Mr. Timothy J    male  54.0      0   
7                     Palsson, Master. Gosta Leonard    male   2.0      3   
8  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  
5      0            330877   8.4583   NaN        Q  
6      0             17463  51.8625   E46        S  
7      1            349909  21.0750   NaN        S  
8      2            347742  11.1333   NaN        S  
9      0            237736  30.0708   NaN        C

# Prepare dataset

We will apply the following transformations on the dataset.

1. Tokenize the names. For example, "Braund, Mr. Owen Harris" will become ["Braund", "Mr.", "Owen", "Harris"].
2. Extract any prefix in the ticket. For example ticket "STON/O2. 3101282" will become "STON/O2." and 3101282.

In [3]:
def preprocess(df):
    df = df.copy()
    
    def normalize_name(x):
        return " ".join([v.strip(",()[].\"'") for v in x.split(" ")])
    
    # takes the last item of the ticket array; should always give an integer
    def ticket_number(x):
        return x.split(" ")[-1]
    
    # creates a new string from the remaining ticket array
    # concatenates if necessary
    # make sure to not return empty strings, but rather NONE
    def ticket_item(x):
        items = x.split(" ")
        if len(items) == 1:
            return "NONE"
        return "_".join(items[0:-1])
    
    df["Name"] = df["Name"].apply(normalize_name)
    df["Ticket_number"] = df["Ticket"].apply(ticket_number)
    df["Ticket_item"] = df["Ticket"].apply(ticket_item)                     
    return df
    
preprocessed_train_df = preprocess(train_df)
preprocessed_serving_df = preprocess(serving_df)

preprocessed_train_df.head(5)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                              Name     Sex   Age  SibSp  \
0                            Braund Mr Owen Harris    male  22.0      1   
1  Cumings Mrs John Bradley Florence Briggs Thayer  female  38.0      1   
2                             Heikkinen Miss Laina  female  26.0      0   
3         Futrelle Mrs Jacques Heath Lily May Peel  female  35.0      1   
4                           Allen Mr William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked Ticket_number Ticket_item  
0      0         A/5 21171   7.2500   NaN        S         21171         A/5  
1      0          PC 17599  71.2833   C85        C         17599          PC  
2      0  STON/O2. 3101282   7.9250   NaN        S       3101282    STON/O2.  
3      0            113803  53.1000  C123        S        113803        NONE  
4      0            373450   8.0500   NaN        S        373450        NONE

Let's keep the list of the input features of the model. Notably, we don't want to train our model on the "PassengerId" and "Ticket" features.

In [4]:
input_features = list(preprocessed_train_df.columns)
input_features.remove("Ticket")
input_features.remove("PassengerId")
input_features.remove("Survived") # this is the outcome
input_features.remove("Ticket_number")

print(f"Input features: {input_features}")

Input features: ['Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Fare', 'Cabin', 'Embarked', 'Ticket_item']


# Convert Pandas dataset to TensorFlow Dataset

In [5]:
def tokenize_names(features, labels=None):
    """Divide the names into tokens. TF-DF can consume text tokens natively."""
    features["Name"] =  tf.strings.split(features["Name"])
    return features, labels

train_ds = tfdf.keras.pd_dataframe_to_tf_dataset(preprocessed_train_df,label="Survived").map(tokenize_names)
serving_ds = tfdf.keras.pd_dataframe_to_tf_dataset(preprocessed_serving_df).map(tokenize_names)

# Train model with default parameters

### Train model

First, we are training a GradientBoostedTreesModel model with the default parameters.

In [6]:
model = tfdf.keras.GradientBoostedTreesModel(
    # let's increase verbosity to get some insights
    verbose=0, # increased to 5
    features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
    exclude_non_specified_features=True, # Only use the features in "features"
    random_seed=1234,
)
model.fit(train_ds)

self_evaluation = model.make_inspector().evaluation()
print(f"Accuracy: {self_evaluation.accuracy} Loss:{self_evaluation.loss}")

[INFO 2024-02-14T14:54:58.722471923+00:00 kernel.cc:1214] Loading model from path /tmp/tmpy9ffdvup/model/ with prefix 4ab2dec8ead04109
[INFO 2024-02-14T14:54:58.732441497+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2024-02-14T14:54:58.732665116+00:00 kernel.cc:1046] Use fast generic engine


Please report this to the TensorFlow team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output.
Cause: could not get source code
To silence this warning, decorate the function with @tf.autograph.experimental.do_not_convert
Accuracy: 0.8152173757553101 Loss:0.8657673597335815


# Train model with improved default parameters

Now you'll use some specific parameters when creating the GBT model

In [7]:
model = tfdf.keras.GradientBoostedTreesModel(
    verbose=1, # Very few logs
    features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
    exclude_non_specified_features=True, # Only use the features in "features"
    
    #num_trees=2000,
    
    # Only for GBT.
    # A bit slower, but great to understand the model.
    compute_permutation_variable_importance=True,
    
    # Change the default hyper-parameters
    hyperparameter_template="benchmark_rank1@v1",
    
    #num_trees=1000,
    #tuner=tuner,
    
    min_examples=1,
    categorical_algorithm="RANDOM",
    max_depth=4, #original disabled
    shrinkage=0.2, #from 0.05)
    #num_candidate_attributes_ratio=0.2,
    split_axis="SPARSE_OBLIQUE",
    sparse_oblique_normalization="MIN_MAX",
    sparse_oblique_num_projections_exponent=2.0,
    num_trees=1400, # increase to 4000 from 2000 -> worse accuracy
    #validation_ratio=0.0,
    random_seed=1234,
    
)
model.fit(train_ds)

self_evaluation = model.make_inspector().evaluation()
print(f"Accuracy: {self_evaluation.accuracy} Loss:{self_evaluation.loss}")

Resolve hyper-parameter template "benchmark_rank1@v1" to "benchmark_rank1@v1" -> {'growing_strategy': 'BEST_FIRST_GLOBAL', 'categorical_algorithm': 'RANDOM', 'split_axis': 'SPARSE_OBLIQUE', 'sparse_oblique_normalization': 'MIN_MAX', 'sparse_oblique_num_projections_exponent': 1.0}.
Use /tmp/tmplv10a9oj as temporary training directory
Reading training dataset...
Training dataset read in 0:00:00.316976. Found 891 examples.
Training model...
Model trained in 0:00:00.360884
Compiling model...


[INFO 2024-02-14T14:55:03.364733908+00:00 decision_forest.cc:661] Model loaded with 12 root(s), 356 node(s), and 9 input feature(s).
[INFO 2024-02-14T14:55:03.36738903+00:00 decision_forest.cc:661] Model loaded with 12 root(s), 356 node(s), and 9 input feature(s).
[INFO 2024-02-14T14:55:03.367668445+00:00 [INFO decision_forest.cc:2024-02-14T14:55:03.367711472+00:00661] Model loaded with  12 root(s), decision_forest.cc:661] Model loaded with 12 root(s), 356 node(s), and 9 input feature(s).356 node(s), and 9 input feature(s).

[INFO 2024-02-14T14:55:03.368701837+00:00 decision_forest.cc:661] Model loaded with 12 root(s), 356 node(s), and 9 input feature(s).
[INFO 2024-02-14T14:55:03.368911215+00:00 decision_forest.cc:661] Model loaded with 12 root(s), 356 node(s), and 9 input feature(s).
[INFO 2024-02-14T14:55:03.36896939+00:00 decision_forest.cc:661] Model loaded with 12 root(s), 356 node(s), and 9 input feature(s).
[[[[INFOINFO  INFO 2024-02-14T14:55:03.370020821+00:002024-02-14T14:55:

Model compiled.
Accuracy: 0.782608687877655 Loss:0.958136260509491


Let's look at the model and you can also notice the information about variable importance that the model figured out

In [8]:
model.summary()

Model: "gradient_boosted_trees_model_1"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
Total params: 1
Trainable params: 0
Non-trainable params: 1
_________________________________________________________________
Type: "GRADIENT_BOOSTED_TREES"
Task: CLASSIFICATION
Label: "__LABEL"

Input Features (10):
	Age
	Cabin
	Embarked
	Fare
	Name
	Parch
	Pclass
	Sex
	SibSp
	Ticket_item

No weights

Variable Importance: INV_MEAN_MIN_DEPTH:
    1.         "Sex"  0.470588 ################
    2.         "Age"  0.350255 ########
    3.        "Fare"  0.333764 #######
    4.        "Name"  0.242640 ##
    5. "Ticket_item"  0.214788 
    6.    "Embarked"  0.205660 
    7.       "Parch"  0.204874 

Variable Importance: MEAN_DECREASE_IN_ACCURACY:
    1.         "Sex"  0.152174 ################
    2.        "Fare"  0.119565 #############
    3.        "Name"  0.032609 #####
    4.       "Parch"  0.010870 ###
    5.      

# Make predictions

In [9]:
def prediction_to_kaggle_format(model, threshold=0.5):
    proba_survive = model.predict(serving_ds, verbose=0)[:,0]
    return pd.DataFrame({
        "PassengerId": serving_df["PassengerId"],
        "Survived": (proba_survive >= threshold).astype(int)
    })

def make_submission(kaggle_predictions):
    path="/kaggle/working/submission.csv"
    kaggle_predictions.to_csv(path, index=False)
    print(f"Submission exported to {path}")
    
kaggle_predictions = prediction_to_kaggle_format(model)
make_submission(kaggle_predictions)
!head /kaggle/working/submission.csv

Submission exported to /kaggle/working/submission.csv
PassengerId,Survived
892,0
893,0
894,0
895,0
896,0
897,0
898,0
899,0
900,1


# Training a model with hyperparameter tunning

Hyper-parameter tuning is enabled by specifying the tuner constructor argument of the model. The tuner object contains all the configuration of the tuner (search space, optimizer, trial and objective).


In [10]:
tuner = tfdf.tuner.RandomSearch(num_trials=1000)
tuner.choice("min_examples", [2, 5, 7, 10])
tuner.choice("categorical_algorithm", ["CART", "RANDOM"])

local_search_space = tuner.choice("growing_strategy", ["LOCAL"])
local_search_space.choice("max_depth", [3, 4, 5, 6, 8])

global_search_space = tuner.choice("growing_strategy", ["BEST_FIRST_GLOBAL"], merge=True)
global_search_space.choice("max_num_nodes", [16, 32, 64, 128, 256])

#tuner.choice("use_hessian_gain", [True, False])
tuner.choice("shrinkage", [0.02, 0.05, 0.10, 0.15, 0.2])
tuner.choice("num_candidate_attributes_ratio", [0.2, 0.5, 0.9, 1.0])


tuner.choice("split_axis", ["AXIS_ALIGNED"])
oblique_space = tuner.choice("split_axis", ["SPARSE_OBLIQUE"], merge=True)
oblique_space.choice("sparse_oblique_normalization",
                     ["NONE", "STANDARD_DEVIATION", "MIN_MAX"])
oblique_space.choice("sparse_oblique_weights", ["BINARY", "CONTINUOUS"])
oblique_space.choice("sparse_oblique_num_projections_exponent", [1.0, 1.5])

# Tune the model. Notice the `tuner=tuner`.
tuned_model = tfdf.keras.GradientBoostedTreesModel(tuner=tuner)
tuned_model.fit(train_ds, verbose=1)

tuned_self_evaluation = tuned_model.make_inspector().evaluation()
print(f"Accuracy: {tuned_self_evaluation.accuracy} Loss:{tuned_self_evaluation.loss}")

Use /tmp/tmpgnreeysa as temporary training directory
Reading training dataset...
Training dataset read in 0:00:00.454619. Found 891 examples.
Training model...
Model trained in 0:01:54.560601
Compiling model...


[INFO 2024-02-14T14:57:00.370535027+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T14:57:00.389175195+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T14:57:00.389236212+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T14:57:00.389270176+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
Accuracy: 0.8767123222351074 Loss:0.6485145092010498


In the last line in the cell above, you can see the accuracy is higher than previously with default parameters and parameters set by hand.

This is the main idea behing hyperparameter tuning.

For more information you can follow this tutorial: [Automated hyper-parameter tuning](https://www.tensorflow.org/decision_forests/tutorials/automatic_tuning_colab)

In [11]:
tuned_model

# Making an ensemble

Here you'll create 100 models with different seeds and combine their results

This approach removes a little bit the random aspects related to creating ML models

In the GBT creation is used the `honest` parameter. It will use different training examples to infer the structure and the leaf values. This regularization technique trades examples for bias estimates.

In [12]:
predictions = None
num_predictions = 0
model = tuned_model # let's try this with the tuned model, seems odd that here it starts over without the tuned model?

for i in range(100):
    print(f"i:{i}")
    # Possible models: GradientBoostedTreesModel or RandomForestModel
    #model = tfdf.keras.GradientBoostedTreesModel(
    #    verbose=0, # Very few logs
    #    features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
    #    exclude_non_specified_features=True, # Only use the features in "features"

        #min_examples=1,
        #categorical_algorithm="RANDOM",
        ##max_depth=4,
        #shrinkage=0.05,
        ##num_candidate_attributes_ratio=0.2,
        #split_axis="SPARSE_OBLIQUE",
        #sparse_oblique_normalization="MIN_MAX",
        #sparse_oblique_num_projections_exponent=2.0,
        #num_trees=2000,
        ##validation_ratio=0.0,
    #    random_seed=i,
    #    honest=True,
    #)
    
    model.fit(train_ds)
    
    sub_predictions = model.predict(serving_ds, verbose=0)[:,0]
    if predictions is None:
        predictions = sub_predictions
    else:
        predictions += sub_predictions
    num_predictions += 1

predictions/=num_predictions

kaggle_predictions = pd.DataFrame({
        "PassengerId": serving_df["PassengerId"],
        "Survived": (predictions >= 0.5).astype(int)
    })

make_submission(kaggle_predictions)

i:0
Reading training dataset...
Training dataset read in 0:00:00.053579. Found 891 examples.
Training model...
Model trained in 0:01:55.199081
Compiling model...


[INFO 2024-02-14T14:58:56.221642833+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T14:58:56.234293927+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T14:58:56.234340362+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T14:58:56.234382391+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:1
Reading training dataset...
Training dataset read in 0:00:00.053041. Found 891 examples.
Training model...
Model trained in 0:01:54.765929
Compiling model...


[INFO 2024-02-14T15:00:51.512074042+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:00:51.525891121+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:00:51.525936824+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:00:51.525967105+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:2
Reading training dataset...
Training dataset read in 0:00:00.046016. Found 891 examples.
Training model...
Model trained in 0:01:55.406412
Compiling model...


[INFO 2024-02-14T15:02:47.473490571+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:02:47.491223074+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:02:47.491310707+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:02:47.491357657+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:3
Reading training dataset...
Training dataset read in 0:00:00.052002. Found 891 examples.
Training model...
Model trained in 0:01:54.368175
Compiling model...


[INFO 2024-02-14T15:04:42.411551535+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:04:42.425736394+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:04:42.425787063+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:04:42.425815352+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:4
Reading training dataset...
Training dataset read in 0:00:00.059098. Found 891 examples.
Training model...
Model trained in 0:01:54.676376
Compiling model...


[INFO 2024-02-14T15:06:37.754560948+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:06:37.770933944+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:06:37.771342084+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:06:37.771416725+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:5
Reading training dataset...
Training dataset read in 0:00:00.059435. Found 891 examples.
Training model...
Model trained in 0:01:55.198442
Compiling model...


[INFO 2024-02-14T15:08:33.57219101+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:08:33.589582794+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:08:33.589673028+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:08:33.589720137+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:6
Reading training dataset...
Training dataset read in 0:00:00.053260. Found 891 examples.
Training model...
Model trained in 0:01:54.649598
Compiling model...


[INFO 2024-02-14T15:10:28.867904297+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:10:28.880400356+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:10:28.880468484+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:10:28.880509376+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:7
Reading training dataset...
Training dataset read in 0:00:00.043039. Found 891 examples.
Training model...
Model trained in 0:01:54.898678
Compiling model...


[INFO 2024-02-14T15:12:24.297143651+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:12:24.308986427+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:12:24.309032021+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:12:24.309058139+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:8
Reading training dataset...
Training dataset read in 0:00:00.044644. Found 891 examples.
Training model...
Model trained in 0:01:54.904614
Compiling model...


[INFO 2024-02-14T15:14:46.419253815+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:14:46.432907279+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:14:46.432953827+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:14:46.432979814+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:9
Reading training dataset...
Training dataset read in 0:00:00.046581. Found 891 examples.
Training model...
Model trained in 0:01:54.234527
Compiling model...


[INFO 2024-02-14T15:17:07.86915184+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:17:07.88567918+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:17:07.885742627+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:17:07.88577481+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:10
Reading training dataset...
Training dataset read in 0:00:00.051104. Found 891 examples.
Training model...
Model trained in 0:01:55.230778
Compiling model...


[INFO 2024-02-14T15:19:03.686136146+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:19:03.700557754+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:19:03.700620994+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:19:03.700651935+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:11
Reading training dataset...
Training dataset read in 0:00:00.051646. Found 891 examples.
Training model...
Model trained in 0:01:54.612725
Compiling model...


[INFO 2024-02-14T15:21:25.191676895+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:21:25.206497456+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:21:25.206580766+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:21:25.206625904+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:12
Reading training dataset...
Training dataset read in 0:00:00.049821. Found 891 examples.
Training model...
Model trained in 0:01:55.156260
Compiling model...


[INFO 2024-02-14T15:23:20.9651545+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:23:20.981718325+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:23:20.981798958+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:23:20.981844082+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:13
Reading training dataset...
Training dataset read in 0:00:00.042811. Found 891 examples.
Training model...
Model trained in 0:01:54.859570
Compiling model...


[INFO 2024-02-14T15:25:16.444323323+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:25:16.457564762+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:25:16.457611303+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:25:16.457642537+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:14
Reading training dataset...
Training dataset read in 0:00:00.049486. Found 891 examples.
Training model...
Model trained in 0:01:55.318636
Compiling model...


[INFO 2024-02-14T15:27:12.313687248+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:27:12.325834236+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:27:12.325879306+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:27:12.325910389+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:15
Reading training dataset...
Training dataset read in 0:00:00.054214. Found 891 examples.
Training model...
Model trained in 0:01:54.488427
Compiling model...


[INFO 2024-02-14T15:29:07.405442856+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:29:07.420528336+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:29:07.420614411+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:29:07.420658746+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:16
Reading training dataset...
Training dataset read in 0:00:00.050837. Found 891 examples.
Training model...
Model trained in 0:01:55.687544
Compiling model...


[INFO 2024-02-14T15:31:03.633279694+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:31:03.64927248+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:31:03.649334812+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:31:03.649364452+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:17
Reading training dataset...
Training dataset read in 0:00:00.043444. Found 891 examples.
Training model...
Model trained in 0:01:54.466108
Compiling model...


[INFO 2024-02-14T15:33:24.501543316+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:33:24.515531135+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:33:24.515579667+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:33:24.515607078+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:18
Reading training dataset...
Training dataset read in 0:00:00.046894. Found 891 examples.
Training model...
Model trained in 0:01:55.457293
Compiling model...


[INFO 2024-02-14T15:35:20.51670376+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:35:20.529439407+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:35:20.529505134+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:35:20.529543594+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:19
Reading training dataset...
Training dataset read in 0:00:00.053739. Found 891 examples.
Training model...
Model trained in 0:01:55.010661
Compiling model...


[INFO 2024-02-14T15:37:16.102021429+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:37:16.115003165+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:37:16.115055454+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:37:16.115084904+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:20
Reading training dataset...
Training dataset read in 0:00:00.043922. Found 891 examples.
Training model...
Model trained in 0:01:55.828144
Compiling model...


[INFO 2024-02-14T15:39:39.019651886+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:39:39.032417073+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:39:39.032479895+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:39:39.032518055+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:21
Reading training dataset...
Training dataset read in 0:00:00.053647. Found 891 examples.
Training model...
Model trained in 0:01:55.462888
Compiling model...


[INFO 2024-02-14T15:41:35.188985042+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:41:35.20663759+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:41:35.20672295+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:41:35.206774828+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:22
Reading training dataset...
Training dataset read in 0:00:00.056463. Found 891 examples.
Training model...
Model trained in 0:01:55.490319
Compiling model...


[INFO 2024-02-14T15:43:57.32454215+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:43:57.337255206+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:43:57.337331966+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:43:57.337391813+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:23
Reading training dataset...
Training dataset read in 0:00:00.050941. Found 891 examples.
Training model...
Model trained in 0:01:55.115729
Compiling model...


[INFO 2024-02-14T15:45:53.027505081+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:45:53.040124865+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:45:53.040177992+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:45:53.040212165+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:24
Reading training dataset...
Training dataset read in 0:00:00.053223. Found 891 examples.
Training model...
Model trained in 0:01:56.127709
Compiling model...


[INFO 2024-02-14T15:47:49.790480586+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:47:49.804991899+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:47:49.805087193+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:47:49.805134918+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:25
Reading training dataset...
Training dataset read in 0:00:00.044927. Found 891 examples.
Training model...
Model trained in 0:01:54.702522
Compiling model...


[INFO 2024-02-14T15:49:45.093271706+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:49:45.108920958+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:49:45.108987522+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:49:45.109036479+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:26
Reading training dataset...
Training dataset read in 0:00:00.052389. Found 891 examples.
Training model...
Model trained in 0:01:55.544082
Compiling model...


[INFO 2024-02-14T15:51:41.194129286+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:51:41.206685116+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:51:41.206732373+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:51:41.206766349+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:27
Reading training dataset...
Training dataset read in 0:00:00.044545. Found 891 examples.
Training model...
Model trained in 0:01:54.414442
Compiling model...


[INFO 2024-02-14T15:53:36.16224072+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:53:36.17398061+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:53:36.17402824+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:53:36.174054565+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:28
Reading training dataset...
Training dataset read in 0:00:00.042847. Found 891 examples.
Training model...
Model trained in 0:01:55.352615
Compiling model...


[INFO 2024-02-14T15:55:32.071749565+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:55:32.085209262+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:55:32.085254647+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:55:32.085280998+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:29
Reading training dataset...
Training dataset read in 0:00:00.048177. Found 891 examples.
Training model...
Model trained in 0:01:54.945536
Compiling model...


[INFO 2024-02-14T15:57:27.754625855+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:57:27.774844643+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:57:27.774906989+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:57:27.77494123+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:30
Reading training dataset...
Training dataset read in 0:00:00.046986. Found 891 examples.
Training model...
Model trained in 0:01:55.431201
Compiling model...


[INFO 2024-02-14T15:59:23.824168342+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T15:59:23.839605619+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T15:59:23.839738289+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T15:59:23.83977843+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:31
Reading training dataset...
Training dataset read in 0:00:00.045262. Found 891 examples.
Training model...
Model trained in 0:01:55.412776
Compiling model...


[INFO 2024-02-14T16:01:19.836921776+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:01:19.852341547+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:01:19.852416101+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:01:19.852448728+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:32
Reading training dataset...
Training dataset read in 0:00:00.049604. Found 891 examples.
Training model...
Model trained in 0:01:55.565011
Compiling model...


[INFO 2024-02-14T16:03:42.123028175+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:03:42.136402785+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:03:42.136469431+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:03:42.136517266+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:33
Reading training dataset...
Training dataset read in 0:00:00.059602. Found 891 examples.
Training model...
Model trained in 0:01:54.644152
Compiling model...


[INFO 2024-02-14T16:05:37.448414304+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:05:37.464472893+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:05:37.464520268+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:05:37.464551378+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:34
Reading training dataset...
Training dataset read in 0:00:00.053388. Found 891 examples.
Training model...
Model trained in 0:01:55.290181
Compiling model...


[INFO 2024-02-14T16:07:33.393925574+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:07:33.406330612+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:07:33.406389982+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:07:33.406416951+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:35
Reading training dataset...
Training dataset read in 0:00:00.044273. Found 891 examples.
Training model...
Model trained in 0:01:54.664134
Compiling model...


[INFO 2024-02-14T16:09:28.623252815+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:09:28.637965685+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:09:28.638027957+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:09:28.638065348+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:36
Reading training dataset...
Training dataset read in 0:00:00.043703. Found 891 examples.
Training model...
Model trained in 0:01:55.272342
Compiling model...


[INFO 2024-02-14T16:11:51.356612955+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:11:51.370928047+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:11:51.371024056+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:11:51.371082376+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:37
Reading training dataset...
Training dataset read in 0:00:00.058893. Found 891 examples.
Training model...
Model trained in 0:01:54.603656
Compiling model...


[INFO 2024-02-14T16:13:46.590245907+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:13:46.60582138+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:13:46.606180713+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:13:46.606511483+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:38
Reading training dataset...
Training dataset read in 0:00:00.053183. Found 891 examples.
Training model...
Model trained in 0:01:55.400384
Compiling model...


[INFO 2024-02-14T16:15:42.553084393+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:15:42.568938237+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:15:42.569033006+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:15:42.569084608+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:39
Reading training dataset...
Training dataset read in 0:00:00.043590. Found 891 examples.
Training model...
Model trained in 0:01:56.764585
Compiling model...


[INFO 2024-02-14T16:17:39.990332208+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:17:40.004150019+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:17:40.004470606+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:17:40.00452346+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:40
Reading training dataset...
Training dataset read in 0:00:00.052144. Found 891 examples.
Training model...
Model trained in 0:01:56.185586
Compiling model...


[INFO 2024-02-14T16:20:01.543663961+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:20:01.556601406+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:20:01.556647806+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:20:01.556674586+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:41
Reading training dataset...
Training dataset read in 0:00:00.059680. Found 891 examples.
Training model...
Model trained in 0:01:56.043312
Compiling model...


[INFO 2024-02-14T16:21:58.227494497+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:21:58.244114292+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:21:58.244439796+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:21:58.244472872+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:42
Reading training dataset...
Training dataset read in 0:00:00.057848. Found 891 examples.
Training model...
Model trained in 0:01:53.882640
Compiling model...


[INFO 2024-02-14T16:24:18.167853877+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:24:18.182441126+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:24:18.182526961+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:24:18.1825765+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:43
Reading training dataset...
Training dataset read in 0:00:00.044017. Found 891 examples.
Training model...
Model trained in 0:01:53.340014
Compiling model...


[INFO 2024-02-14T16:26:12.138712621+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:26:12.153445689+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:26:12.15350945+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:26:12.153540418+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:44
Reading training dataset...
Training dataset read in 0:00:00.055606. Found 891 examples.
Training model...
Model trained in 0:01:55.567217
Compiling model...


[INFO 2024-02-14T16:28:08.395108372+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:28:08.406540424+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:28:08.406584821+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:28:08.406615204+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:45
Reading training dataset...
Training dataset read in 0:00:00.057779. Found 891 examples.
Training model...
Model trained in 0:01:56.147130
Compiling model...


[INFO 2024-02-14T16:30:31.068581002+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:30:31.091646102+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:30:31.09171878+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:30:31.091753295+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:46
Reading training dataset...
Training dataset read in 0:00:00.048511. Found 891 examples.
Training model...
Model trained in 0:01:56.430067
Compiling model...


[INFO 2024-02-14T16:32:28.220354096+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:32:28.232925181+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:32:28.232976171+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:32:28.233008619+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:47
Reading training dataset...
Training dataset read in 0:00:00.052601. Found 891 examples.
Training model...
Model trained in 0:01:55.619427
Compiling model...


[INFO 2024-02-14T16:34:24.445666571+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:34:24.457589505+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:34:24.457638971+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:34:24.457667943+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:48
Reading training dataset...
Training dataset read in 0:00:00.053945. Found 891 examples.
Training model...
Model trained in 0:01:54.838158
Compiling model...


[INFO 2024-02-14T16:36:19.846238818+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:36:19.859748698+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:36:19.859807367+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:36:19.859836387+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:49
Reading training dataset...
Training dataset read in 0:00:00.046101. Found 891 examples.
Training model...
Model trained in 0:01:56.028670
Compiling model...


[INFO 2024-02-14T16:38:16.495430009+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:38:16.518570616+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:38:16.518639301+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:38:16.518671797+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:50
Reading training dataset...
Training dataset read in 0:00:00.044240. Found 891 examples.
Training model...
Model trained in 0:01:55.405743
Compiling model...


[INFO 2024-02-14T16:40:12.557142109+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:40:12.572561041+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:40:12.572627514+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:40:12.572661453+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:51
Reading training dataset...
Training dataset read in 0:00:00.044768. Found 891 examples.
Training model...
Model trained in 0:01:55.107812
Compiling model...


[INFO 2024-02-14T16:42:34.354557593+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:42:34.369706543+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:42:34.369776433+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:42:34.369809194+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:52
Reading training dataset...
Training dataset read in 0:00:00.049385. Found 891 examples.
Training model...
Model trained in 0:01:54.827907
Compiling model...


[INFO 2024-02-14T16:44:56.169849269+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:44:56.185233615+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:44:56.185295042+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:44:56.185325204+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:53
Reading training dataset...
Training dataset read in 0:00:00.044072. Found 891 examples.
Training model...
Model trained in 0:01:55.062332
Compiling model...


[INFO 2024-02-14T16:46:51.895627246+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:46:51.917685712+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:46:51.917755437+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:46:51.917790385+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:54
Reading training dataset...
Training dataset read in 0:00:00.044095. Found 891 examples.
Training model...
Model trained in 0:01:54.067997
Compiling model...


[INFO 2024-02-14T16:48:46.68108315+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:48:46.69360913+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:48:46.693654873+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:48:46.693681794+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:55
Reading training dataset...
Training dataset read in 0:00:00.052099. Found 891 examples.
Training model...
Model trained in 0:01:54.867855
Compiling model...


[INFO 2024-02-14T16:50:42.096045653+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:50:42.110946843+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:50:42.111030179+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:50:42.111078377+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:56
Reading training dataset...
Training dataset read in 0:00:00.041980. Found 891 examples.
Training model...
Model trained in 0:01:53.909456
Compiling model...


[INFO 2024-02-14T16:52:36.590215292+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:52:36.603667518+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:52:36.603744715+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:52:36.60378874+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:57
Reading training dataset...
Training dataset read in 0:00:00.044984. Found 891 examples.
Training model...
Model trained in 0:01:54.853126
Compiling model...


[INFO 2024-02-14T16:54:32.147729677+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:54:32.164795816+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:54:32.164857571+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:54:32.164898615+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:58
Reading training dataset...
Training dataset read in 0:00:00.059116. Found 891 examples.
Training model...
Model trained in 0:01:52.818347
Compiling model...


[INFO 2024-02-14T16:56:52.226973141+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:56:52.23949539+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:56:52.239599243+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:56:52.239642689+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:59
Reading training dataset...
Training dataset read in 0:00:00.044365. Found 891 examples.
Training model...
Model trained in 0:01:53.242060
Compiling model...


[INFO 2024-02-14T16:59:14.749539258+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T16:59:14.763648008+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T16:59:14.763730049+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T16:59:14.76377467+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:60
Reading training dataset...
Training dataset read in 0:00:00.072130. Found 891 examples.
Training model...
Model trained in 0:01:55.856257
Compiling model...


[INFO 2024-02-14T17:01:39.519625514+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:01:39.534791635+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:01:39.534868165+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:01:39.534909149+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:61
Reading training dataset...
Training dataset read in 0:00:00.056397. Found 891 examples.
Training model...
Model trained in 0:01:55.276259
Compiling model...


[INFO 2024-02-14T17:03:35.441631516+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:03:35.465472653+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:03:35.465543909+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:03:35.465579454+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:62
Reading training dataset...
Training dataset read in 0:00:00.057473. Found 891 examples.
Training model...
Model trained in 0:01:54.656533
Compiling model...


[INFO 2024-02-14T17:05:30.810562937+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:05:30.826810696+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:05:30.826903423+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:05:30.826952432+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:63
Reading training dataset...
Training dataset read in 0:00:00.043809. Found 891 examples.
Training model...
Model trained in 0:01:54.612598
Compiling model...


[INFO 2024-02-14T17:07:26.07810063+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:07:26.090710064+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:07:26.090760605+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:07:26.090789849+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:64
Reading training dataset...
Training dataset read in 0:00:00.049644. Found 891 examples.
Training model...
Model trained in 0:01:54.771564
Compiling model...


[INFO 2024-02-14T17:09:48.362323851+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:09:48.37578844+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:09:48.375840641+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:09:48.375868662+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:65
Reading training dataset...
Training dataset read in 0:00:00.045693. Found 891 examples.
Training model...
Model trained in 0:01:54.689593
Compiling model...


[INFO 2024-02-14T17:11:43.731699992+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:11:43.748947793+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:11:43.748998564+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:11:43.749027283+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:66
Reading training dataset...
Training dataset read in 0:00:00.043175. Found 891 examples.
Training model...
Model trained in 0:01:54.047907
Compiling model...


[INFO 2024-02-14T17:13:38.353568654+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:13:38.366315316+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:13:38.366362056+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:13:38.366410979+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:67
Reading training dataset...
Training dataset read in 0:00:00.052810. Found 891 examples.
Training model...
Model trained in 0:01:54.965524
Compiling model...


[INFO 2024-02-14T17:15:33.900498533+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:15:33.914299594+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:15:33.914394681+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:15:33.914448968+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:68
Reading training dataset...
Training dataset read in 0:00:00.042911. Found 891 examples.
Training model...
Model trained in 0:01:54.859030
Compiling model...


[INFO 2024-02-14T17:17:55.873914351+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:17:55.88824444+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:17:55.888295322+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:17:55.888325219+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:69
Reading training dataset...
Training dataset read in 0:00:00.054423. Found 891 examples.
Training model...
Model trained in 0:01:54.867228
Compiling model...


[INFO 2024-02-14T17:19:51.481220668+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:19:51.50341664+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:19:51.503508324+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:19:51.50355788+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:70
Reading training dataset...
Training dataset read in 0:00:00.052599. Found 891 examples.
Training model...
Model trained in 0:01:55.135570
Compiling model...


[INFO 2024-02-14T17:21:47.280494468+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:21:47.296119315+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:21:47.296204614+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:21:47.296253862+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:71
Reading training dataset...
Training dataset read in 0:00:00.048876. Found 891 examples.
Training model...
Model trained in 0:01:55.345662
Compiling model...


[INFO 2024-02-14T17:23:43.30928244+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:23:43.321035926+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:23:43.321080551+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:23:43.321110442+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:72
Reading training dataset...
Training dataset read in 0:00:00.043220. Found 891 examples.
Training model...
Model trained in 0:01:54.206939
Compiling model...


[INFO 2024-02-14T17:25:38.091356855+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:25:38.10489456+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:25:38.104956108+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:25:38.104993189+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:73
Reading training dataset...
Training dataset read in 0:00:00.043058. Found 891 examples.
Training model...
Model trained in 0:01:54.556412
Compiling model...


[INFO 2024-02-14T17:27:33.233529237+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:27:33.254853416+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:27:33.254913982+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:27:33.254948553+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:74
Reading training dataset...
Training dataset read in 0:00:00.053942. Found 891 examples.
Training model...
Model trained in 0:01:54.774360
Compiling model...


[INFO 2024-02-14T17:29:55.575179049+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:29:55.587757576+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:29:55.587835534+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:29:55.587874655+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:75
Reading training dataset...
Training dataset read in 0:00:00.046927. Found 891 examples.
Training model...
Model trained in 0:01:53.476172
Compiling model...


[INFO 2024-02-14T17:31:49.653521346+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:31:49.66512257+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:31:49.665183762+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:31:49.665221307+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:76
Reading training dataset...
Training dataset read in 0:00:00.042437. Found 891 examples.
Training model...
Model trained in 0:01:52.970055
Compiling model...


[INFO 2024-02-14T17:33:43.154760351+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:33:43.16839064+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:33:43.168454316+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:33:43.168494616+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:77
Reading training dataset...
Training dataset read in 0:00:00.054297. Found 891 examples.
Training model...
Model trained in 0:01:55.268587
Compiling model...


[INFO 2024-02-14T17:36:07.590900851+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:36:07.612307352+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:36:07.612520047+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:36:07.612589573+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:78
Reading training dataset...
Training dataset read in 0:00:00.047404. Found 891 examples.
Training model...
Model trained in 0:01:55.175390
Compiling model...


[INFO 2024-02-14T17:38:03.440605792+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:38:03.454137441+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:38:03.454213487+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:38:03.454254627+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:79
Reading training dataset...
Training dataset read in 0:00:00.044546. Found 891 examples.
Training model...
Model trained in 0:01:54.748310
Compiling model...


[INFO 2024-02-14T17:39:58.8296387+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:39:58.845015314+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:39:58.845077923+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:39:58.845112365+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:80
Reading training dataset...
Training dataset read in 0:00:00.047952. Found 891 examples.
Training model...
Model trained in 0:01:54.685264
Compiling model...


[INFO 2024-02-14T17:41:54.116542462+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:41:54.133108328+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:41:54.133620646+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:41:54.133680037+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:81
Reading training dataset...
Training dataset read in 0:00:00.054214. Found 891 examples.
Training model...
Model trained in 0:01:54.000853
Compiling model...


[INFO 2024-02-14T17:43:48.781502766+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:43:48.797265494+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:43:48.797314906+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:43:48.797342922+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:82
Reading training dataset...
Training dataset read in 0:00:00.045045. Found 891 examples.
Training model...
Model trained in 0:01:55.518410
Compiling model...


[INFO 2024-02-14T17:46:12.389103552+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:46:12.401444009+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:46:12.401492758+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:46:12.401518803+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:83
Reading training dataset...
Training dataset read in 0:00:00.068745. Found 891 examples.
Training model...
Model trained in 0:01:54.932156
Compiling model...


[INFO 2024-02-14T17:48:33.926267312+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:48:33.942824656+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:48:33.942895481+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:48:33.942935847+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:84
Reading training dataset...
Training dataset read in 0:00:00.045079. Found 891 examples.
Training model...
Model trained in 0:01:55.422404
Compiling model...


[INFO 2024-02-14T17:50:56.529068831+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:50:56.542338445+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:50:56.542399741+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:50:56.542431875+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:85
Reading training dataset...
Training dataset read in 0:00:00.051407. Found 891 examples.
Training model...
Model trained in 0:01:54.442277
Compiling model...


[INFO 2024-02-14T17:52:51.676113891+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:52:51.693487524+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:52:51.693533364+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:52:51.693566072+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:86
Reading training dataset...
Training dataset read in 0:00:00.052392. Found 891 examples.
Training model...
Model trained in 0:01:54.589686
Compiling model...


[INFO 2024-02-14T17:54:46.907743951+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:54:46.920420262+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:54:46.920471889+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:54:46.920504449+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:87
Reading training dataset...
Training dataset read in 0:00:00.047184. Found 891 examples.
Training model...
Model trained in 0:01:54.604871
Compiling model...


[INFO 2024-02-14T17:56:42.15363294+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:56:42.165325655+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:56:42.165388618+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:56:42.165420904+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:88
Reading training dataset...
Training dataset read in 0:00:00.049890. Found 891 examples.
Training model...
Model trained in 0:01:54.517961
Compiling model...


[INFO 2024-02-14T17:58:37.2951038+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T17:58:37.310564633+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T17:58:37.310646559+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T17:58:37.310692442+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:89
Reading training dataset...
Training dataset read in 0:00:00.053667. Found 891 examples.
Training model...
Model trained in 0:01:55.872211
Compiling model...


[INFO 2024-02-14T18:00:33.885344832+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T18:00:33.921189799+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T18:00:33.921264496+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T18:00:33.921310817+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:90
Reading training dataset...
Training dataset read in 0:00:00.053964. Found 891 examples.
Training model...
Model trained in 0:01:55.001401
Compiling model...


[INFO 2024-02-14T18:02:29.811082723+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T18:02:29.834206212+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T18:02:29.834258627+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T18:02:29.834292257+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:91
Reading training dataset...
Training dataset read in 0:00:00.047797. Found 891 examples.
Training model...
Model trained in 0:01:54.401039
Compiling model...


[INFO 2024-02-14T18:04:24.893595927+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T18:04:24.913653622+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T18:04:24.913703415+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T18:04:24.913735481+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:92
Reading training dataset...
Training dataset read in 0:00:00.059821. Found 891 examples.
Training model...
Model trained in 0:01:55.650240
Compiling model...


[INFO 2024-02-14T18:06:21.243229925+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T18:06:21.297603736+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T18:06:21.297687876+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T18:06:21.297737265+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:93
Reading training dataset...
Training dataset read in 0:00:00.061168. Found 891 examples.
Training model...
Model trained in 0:01:54.712372
Compiling model...


[INFO 2024-02-14T18:08:16.85764595+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T18:08:16.885920767+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T18:08:16.88598848+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T18:08:16.886021975+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:94
Reading training dataset...
Training dataset read in 0:00:00.049627. Found 891 examples.
Training model...
Model trained in 0:01:54.600662
Compiling model...


[INFO 2024-02-14T18:10:12.110537726+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T18:10:12.13381995+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T18:10:12.133884342+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T18:10:12.133922639+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:95
Reading training dataset...
Training dataset read in 0:00:00.059811. Found 891 examples.
Training model...
Model trained in 0:01:54.166135
Compiling model...


[INFO 2024-02-14T18:12:33.845505158+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T18:12:33.867699623+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T18:12:33.868035435+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T18:12:33.868269569+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:96
Reading training dataset...
Training dataset read in 0:00:00.053984. Found 891 examples.
Training model...
Model trained in 0:01:55.232235
Compiling model...


[INFO 2024-02-14T18:14:29.936214555+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T18:14:29.967143719+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T18:14:29.967243352+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T18:14:29.967298811+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:97
Reading training dataset...
Training dataset read in 0:00:00.048385. Found 891 examples.
Training model...
Model trained in 0:01:54.735432
Compiling model...


[INFO 2024-02-14T18:16:51.547949616+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T18:16:51.56692204+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T18:16:51.566974515+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T18:16:51.567003947+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:98
Reading training dataset...
Training dataset read in 0:00:00.044291. Found 891 examples.
Training model...
Model trained in 0:01:55.698891
Compiling model...


[INFO 2024-02-14T18:18:47.94526599+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T18:18:47.962042054+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T18:18:47.962108648+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T18:18:47.962147369+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
i:99
Reading training dataset...
Training dataset read in 0:00:00.058430. Found 891 examples.
Training model...
Model trained in 0:01:55.156858
Compiling model...


[INFO 2024-02-14T18:20:43.728499241+00:00 kernel.cc:1214] Loading model from path /tmp/tmpgnreeysa/model/ with prefix 88ec59f00d304709
[INFO 2024-02-14T18:20:43.77355532+00:00 decision_forest.cc:661] Model loaded with 16 root(s), 452 node(s), and 12 input feature(s).
[INFO 2024-02-14T18:20:43.773624167+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2024-02-14T18:20:43.773660662+00:00 kernel.cc:1046] Use fast generic engine


Model compiled.
Submission exported to /kaggle/working/submission.csv


# What is next

If you want to learn more about TensorFlow Decision Forests and its advanced features, you can follow the official documentation [here](https://www.tensorflow.org/decision_forests) 